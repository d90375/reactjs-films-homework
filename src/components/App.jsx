import React from "react";
import { BrowserRouter as Router, Switch, Route, Redirect } from "react-router-dom";
import NotFound from "../pages/NotFound";
import MainPage from "../pages/MainPage";

const App = () => {
  return (
    <div className="wrap">
      <Router>
        <Switch>
          <Redirect exact from="/" to="/film/734309?filter=trending" />
          <Route path="/film/:id" component={MainPage} />
          <Route component={NotFound} />
        </Switch>
      </Router>
    </div>
  );
};

export default App;
