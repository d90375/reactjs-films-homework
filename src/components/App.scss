.wrap {
  min-width: 320px;
  max-width: 2400px;
  padding: 0;
  box-sizing: border-box;
  margin: 0 auto;
}

.main {
  width: 2400px;
  background-color: #f5f6f6;
}

.main__wrap {
  padding-top: 71px;
  padding-left: 140px;
  padding-right: 140px;
}
