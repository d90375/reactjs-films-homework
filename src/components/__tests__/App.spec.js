import React from "react";
import App from "../App";

describe("App component", () => {
  let component;

  it("should render App component without props", () => {
    component = shallow(<App />);
    expect(component).toMatchSnapshot();
  });
});
