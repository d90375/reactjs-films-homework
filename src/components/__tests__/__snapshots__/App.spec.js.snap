// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App component without props 1`] = `
<div
  className="wrap"
>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/film/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
